tests.py

```diff
@@ -292,3 +300,22 @@ def test_error(self, get):
                                             'errortext': 'Fail'}}
         cs = CloudStack(endpoint='localhost', key='foo', secret='bar')
         self.assertRaises(CloudStackException, cs.listVirtualMachines)
+
+    @patch("requests.get")
+    def test_signature_v3(self, get):
+        cs = CloudStack(endpoint='localhost', key='foo', secret='bar',
+                        expiration=600)
+        get.return_value.status_code = 200
+        get.return_value.json.return_value = {
+            'createnetworkresponse': {},
+        }
+        cs.createNetwork(name="", display_text="")
+
+        _, kwargs = get.call_args
+        params = kwargs['params']
+        assert '3' == params['signatureVersion'], kwargs
+
+        # we ignore the timezone for Python2's lack of %z
+        expires = datetime.datetime.strptime(params['expires'][:18],
```

Comment:
This leaves only 1 digit for seconds (`len('2018-09-18T10:43:14') = 19`). It'd be more explicit to use an assignment and pass `[:length]` instead of an int.

Reply:
this is a test and a hack for Python2